DFE-Digital/teaching-vacancies
# [TEVA-2509] Track jobs API usage

## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-2509

## Changes in this PR:

- Track jobs API usage using RequestEvent

    We want to monitor usage of the API to see frequency of usage and
    technologies used. We also want to not send irrelevant data to bigquery.
    
    Since we are monitoring API usage we might want to distinguish between
    incorrect and correct usage of the API, so I have allowed failing api
    queries to also trigger events. These contain the key-value pair:
    `not_found: true`

- Fix terms and conditions link in API json

    This link was pointing to a page that does not contain the terms and conditions
    (just a tickbox to confirm you've read the T&C and a link to the T&C), and which
    was hidden behind authenticating as a publisher.

    Now the link points to the T&C page without requiring authentication.

## How to test this locally

Start your sidekiq server so that localhost can send events. You should see the events in BigQuery when you do a relevant query, within a few seconds.

## Reviewing on the review app

Visit the api on the review app:

https://teaching-vacancies-review-pr-4239.london.cloudapps.digital/api/v1/jobs.json

or 

https://teaching-vacancies-review-pr-4239.london.cloudapps.digital/api/v1/jobs/insert-a-job-id-here.json

or generate a not found response:

https://teaching-vacancies-review-pr-4239.london.cloudapps.digital/api/v1/jobs/i-represent-someone-misusing-the-api.json

The event is immediately streamed to BigQuery, where you can look at it by running this query:

```sql
SELECT
  *
FROM
  `teacher-vacancy-service.staging_dataset.events`
WHERE
  DATE(occurred_at) = "2021-11-05"
  AND request_path LIKE '%api%'
ORDER BY
  DATETIME(occurred_at) DESC
LIMIT
  100;
```
